### PR TITLE
Change DeepLinkParameters not to use subEntity* anymore

### PR DESCRIPTION
**Pull request by MarcinJuraszek**

### CODE REVIEW

**Review by jadahiya-MSFT — APPROVED** [review]

**Comment by jadahiya-MSFT** [comment]
Android E2E tests seem to be a bit flaky and might need to be rerun, I've talked with @jekloudaMSFT who is investigating why this is happening.

**Comment by MarcinJuraszek** [comment]
I think in this case it's an actual test failure. I pushed a new commit to try to mitigate that.

**Review by jadahiya-MSFT — DISMISSED** [review]

**Comment by MarcinJuraszek** [comment]
OK. Looks like it wasn't my fault after all :)

**Review by jadahiya-MSFT — DISMISSED** [review]

File: apps/teams-test-app/src/components/PagesAPIs.tsx [hunk 1]
```diff
@@ -95,11 +95,27 @@ const ShareDeepLink = (): ReactElement =>
...
         },
         withCallback: (input, setResult) => {
-          shareDeepLink(input);
+          if (input.subEntityId && input.subEntityLabel) {
+            shareDeepLink(input);
+          } else {
+            shareDeepLink({
+              subEntityId: input.subPageId,
+              subEntityLabel: input.subPageLabel,
+              subEntityWebUrl: input.subPageWebUrl,
```
Comment (erinha) [inline apps/teams-test-app/src/components/PagesAPIs.tsx:116]: Is this intentionally reversed vs the previous change block?

File: packages/teams-js/src/public/interfaces.ts [hunk 2]
```diff
@@ -513,6 +513,29 @@ export interface Context {
...
+  /**
+   * The fallback URL to which to navigate the user if the client cannot render the page.
+   * This URL should lead directly to the sub-entity.
+   */
+  subPageWebUrl?: string;
+}
+
+/**
+ * @deprecated
+ * As of 2.0.0-beta.1, please use {@link ShareDeepLinkParameters} instead.
```
Comment (erinha) [inline packages/teams-js/src/public/interfaces.ts:537]: I think this tag should say "As of 2.0.0-beta.5", right?

**Review by erinha — CHANGES_REQUESTED** [review]
A couple small comments

File: apps/teams-test-app/src/components/PagesAPIs.tsx (hunk 1 above)
Comment (MarcinJuraszek) [reply apps/teams-test-app/src/components/PagesAPIs.tsx:116]: This is calling into the back-compat function which takes old parameters. So yes, this is meant to be the other way around :)

**Review by jadahiya-MSFT — APPROVED** [review]

**Review by erinha — APPROVED** [review]
🚀

**Pull request merged**